src/sentry/spans/consumers/process/factory.py

```diff
@@ -156,50 +104,41 @@ def _process_group_raw(snuba_span: MutableMapping[str, Any], transaction: str) -
         metrics.incr("spans.invalid_group_raw")
 
 
-def _format_event_id(payload: Mapping[str, Any]) -> str:
-    event_id = payload.get("event_id")
+def _format_event_id(payload: Mapping[str, Any], key="event_id") -> Optional[str]:
+    event_id = payload.get(key)
     if event_id:
         return uuid.UUID(event_id).hex
-    return ""
+    return None
 
 
 def _deserialize_payload(payload: bytes) -> Mapping[str, Any]:
-    # We're migrating the payload from being encoded in msgpack to JSON.
-    # This for backward compatibility while we transition.
-    try:
-        return msgpack.unpackb(payload)
-    except ValueError:
-        return json.loads(payload, use_rapid_json=True)
+    return SPAN_SCHEMA.decode(payload)
 
 
 def _process_message(message: Message[KafkaPayload]) -> KafkaPayload:
     payload = _deserialize_payload(message.payload.value)
     relay_span = payload["span"]
+    relay_span["event_id"] = payload.get("event_id")
+    relay_span["organization_id"] = payload["organization_id"]
     relay_span["project_id"] = payload["project_id"]
-    relay_span["event_id"] = _format_event_id(payload)
-
-    organization_id = payload.get("organization_id")
-    retention_days = payload.get("retention_days")
-
-    if not organization_id or not retention_days:
-        organization_id, retention_days = get_organization(
-            relay_span["project_id"],
-        )
-
-    relay_span["organization_id"] = organization_id
-    relay_span["retention_days"] = retention_days
-
-    if "sentry_tags" not in relay_span and "data" in relay_span:
-        relay_span["sentry_tags"] = relay_span.pop("data")
+    relay_span["retention_days"] = payload["retention_days"]
 
-    snuba_span = _process_relay_span_v0(relay_span)
+    snuba_span = _process_relay_span_v1(relay_span)
     snuba_payload = json.dumps(snuba_span).encode("utf-8")
     return KafkaPayload(key=None, value=snuba_payload, headers=[])
 
 
 def process_message(message: Message[KafkaPayload]) -> KafkaPayload:
     try:
         return _process_message(message)
+    except ValidationError as e:
```

Comment:
@lynnagara @untitaker How should I handle that? I want to raise an `InvalidMessage` exception so we skip the message and also log the `ValidationError` to Sentry to know what's happening. Would that be the way to do it?

Reply:
I think the easiest in this situation would be to log manually like you do and `return message.replace(FILTERED_PAYLOAD)`, so the next `Produce` step does not actually produce it (but the offsets still get committed)

```
from arroyo.types import FILTERED_PAYLOAD
```

i think this consumer does not have a DLQ, so raising InvalidMessage will actually stop the consumer

Reply:
What do you mean? Currently, raising `InvalidMessage` seems to do the job. Or is it what's causing the pauses we see often?

Is this `FILTERED_PAYLOAD` just a way for the `Produce` to understand it's an empty payload and we don't want to produce it but we want to commit?

Reply:
> What do you mean? Currently, raising InvalidMessage seems to do the job. Or is it what's causing the pauses we see often?

oh, i think i misremember how it works then. nevermind

InvalidMessage is still not the fastest thing in the world, it is indeed faster to directly transform to FilteredPayload. and yeah I think you have to change the mypy type